### PR TITLE
feat: add attributes for cloud-native deployment

### PR DESCRIPTION
**Pull request by thisthat**
Signed-off-by: Giovanni Liva <giovanni.liva@dynatrace.com>Fixes #

## Changes

Please provide a brief description of the changes here.

For non-trivial changes, follow the [change proposal process](https://github.com/open-telemetry/opentelemetry-specification/blob/main/CONTRIBUTING.md#proposing-a-change) and link to the related issue(s) and/or [OTEP(s)](https://github.com/open-telemetry/oteps), update the [`CHANGELOG.md`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/CHANGELOG.md), and also be sure to update [`spec-compliance-matrix.md`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/spec-compliance-matrix.md) if necessary.

Related issues #

Related [OTEP(s)](https://github.com/open-telemetry/oteps) #

### CODE REVIEW

File: semantic_conventions/trace/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        examples: [ '0.1.0' ]
+      - id: status
+        type:
+          allow_custom_values: true
+          members:
+            - id: 'pending'
+              value: 'Pending'
+              brief: 'The deployment has been accepted and is being setting up'
+            - id: 'progressing'
+              value: 'Progressing'
```
Comment (mowies) [inline semantic_conventions/trace/deployment.yaml:29]: IMHO, "In Progress" sounds much better...

File: semantic_conventions/trace/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+              brief: 'The deployment is taking place'
```
Comment (mowies) [inline semantic_conventions/trace/deployment.yaml]:
```suggestion
              brief: 'The deployment is currently taking place'
```

File: semantic_conventions/trace/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,121 @@
+groups:
+  - id: deployment.application
+    prefix: deployment.application
+    type: event
+    brief: >
+        This document defines attributes for Cloud Native Deployments of Applications.
```
Comment (mowies) [inline semantic_conventions/trace/deployment.yaml]:
```suggestion
        This document defines attributes for Cloud Native deployments of applications.
```

File: semantic_conventions/trace/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        examples: [ 'v0.1.0' ]
+      - id: status
+        type:
+          allow_custom_values: true
+          members:
+            - id: 'pending'
+              value: 'Pending'
+              brief: 'The deployment has been accepted and is being setting up'
+            - id: 'progressing'
+              value: 'Progressing'
```
Comment (mowies) [inline semantic_conventions/trace/deployment.yaml:76]: same thing here...

File: semantic_conventions/trace/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+              brief: 'The deployment is taking place'
```
Comment (mowies) [inline semantic_conventions/trace/deployment.yaml]:
```suggestion
              brief: 'The deployment is currently taking place'
```

File: semantic_conventions/trace/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+      This document defines attributes for accompanying tasks that support Cloud Native deployments.
+    attributes:
+      - ref: deployment.application.name
+      - ref: deployment.workload.name
+        requirement_level: recommended
+      - id: name
+        type: string
+        requirement_level: required
+        brief: >
+          The name that identifies the task executed.
```
Comment (mowies) [inline semantic_conventions/trace/deployment.yaml]:
```suggestion
          The name that identifies the executed task.
```

File: semantic_conventions/trace/deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+              value: 'Succeeded'
+              brief: 'The deployment was completed correctly'
+            - id: 'failed'
+              value: 'Failed'
+              brief: 'The deployment was **not** completed correctly'
+            - id: 'unknown'
+              value: 'Unknown'
+              brief: 'The deployment is unsuccessful and the reason is unknown'
+        brief: >
+          The status of the task execution.
```
Comment (mowies) [inline semantic_conventions/trace/deployment.yaml]: newline :)

File: semantic_conventions/trace/deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+        brief: >
+          The name that identifies the task executed.
+        examples: [ 'database-migration', 'report-metric', 'provision-infrastructure' ]
+      - id: status
+        type:
+          allow_custom_values: true
+          members:
+            - id: 'pending'
+              value: 'Pending'
+              brief: 'The deployment has been accepted and is being setting up'
```
Comment (mowies) [inline semantic_conventions/trace/deployment.yaml]:
```suggestion
              brief: 'The deployment has been accepted and is being set up'
```

File: semantic_conventions/trace/deployment.yaml [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+        brief: >
+          The version of the workload under deployment.
+        examples: [ 'v0.1.0' ]
+      - id: status
+        type:
+          allow_custom_values: true
+          members:
+            - id: 'pending'
+              value: 'Pending'
+              brief: 'The deployment has been accepted and is being setting up'
```
Comment (mowies) [inline semantic_conventions/trace/deployment.yaml]:
```suggestion
              brief: 'The deployment has been accepted and is being set up'
```

File: semantic_conventions/trace/deployment.yaml [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+        brief: >
+          The version of the application under deployment.
+        examples: [ '0.1.0' ]
+      - id: status
+        type:
+          allow_custom_values: true
+          members:
+            - id: 'pending'
+              value: 'Pending'
+              brief: 'The deployment has been accepted and is being setting up'
```
Comment (mowies) [inline semantic_conventions/trace/deployment.yaml]:
```suggestion
              brief: 'The deployment has been accepted and is being set up'
```

File: semantic_conventions/logs/log-deployment.yaml [hunk 11]
```diff
@@ -0,0 +1,37 @@
+groups:
+  - id: log-deployment-application
+    type: event
+    prefix: deployment.application
+    brief: >
+      This document defines attributes for cloud native deployments
```
Comment (mowies) [inline semantic_conventions/logs/log-deployment.yaml]: here, you don't capitalize cloud and native, but in the other file(s), you did...

File: semantic_conventions/trace/deployment.yaml [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+              brief: 'The deployment is taking place'
+            - id: 'succeeded'
+              value: 'Succeeded'
+              brief: 'The deployment was completed correctly'
+            - id: 'failed'
+              value: 'Failed'
+              brief: 'The deployment was **not** completed correctly'
+            - id: 'unknown'
+              value: 'Unknown'
+              brief: 'The deployment is unsuccessful and the reason is unknown'
```
Comment (mowies) [inline semantic_conventions/trace/deployment.yaml]:
```suggestion
              brief: 'The deployment was unsuccessful and the reason is unknown'
```

File: semantic_conventions/trace/deployment.yaml [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
```
Comment (mowies) [inline semantic_conventions/trace/deployment.yaml]:
```suggestion
              brief: 'The deployment was unsuccessful and the reason is unknown'
```

File: semantic_conventions/trace/deployment.yaml [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
```
Comment (mowies) [inline semantic_conventions/trace/deployment.yaml]:
```suggestion
              brief: 'The deployment was unsuccessful and the reason is unknown'
```

**Review by mowies — COMMENTED** [review]
capitalization with [Cc]loud [Nn]ative and [Aa]pplication [Dd]eployments is not always consistent

File: semantic_conventions/trace/deployment.yaml (hunk 1 above)
Comment (thisthat) [reply semantic_conventions/trace/deployment.yaml:29]: I took the wording from the K8s docs: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#progressing-deployment

**Pull request closed without merging**